HW04.Rmd

```diff
@@ -68,54 +126,114 @@ Assume that:
 * Summer = July-September
 * Fall = October-December
 
-```{r season}
-
-```
+**Answer**
+Based on the previous table, summer has the most gun deaths.
 
 
 # These are more open ended questions from here on out, you could look at making multiple summaries or types of plots. 
 
 ### Are whites who are killed by guns more likely to die because of suicide or homicide? How does this compare to blacks and hispanics?
 
 ```{r race}
-
+filtered_race <- gun_deaths %>%
+    filter(race %in% c("White", "Black", "Hispanic"), intent %in% c("Suicide", "Homicide")) %>%
+    group_by(race, intent) %>%
+    count() 
+
+ggplot(filtered_race, mapping = aes(x = race, y = n)) +
+           geom_col(aes(fill = intent)) +
+           labs(title = "Distribution of Death Probability due to Suicide vs. Homicide by Races", x = "Races", y = "Number of Deaths") 
 ```
+**Answer**
+Based on this graph, whites who are killed by guns are more likely to die because of suicide. In other races (black and hispanic), they are more likely to get killed by guns due to homicide.
 
 (hint maybe looking at percentages would be good)
 ```{r race_2}
-
+filtered_race <- gun_deaths %>%
+    filter(race %in% c("White", "Black", "Hispanic"), intent %in% c("Suicide", "Homicide")) %>%
+    group_by(race, intent) %>%
+    count(race) %>%
+    group_by(race) %>%
+    mutate(percentages = 100*n / sum(n))
+
+ggplot(filtered_race, mapping = aes(x = race, y = percentages)) +
+           geom_col(aes(fill = intent)) +
+           labs(title = "Distribution of Death Probability due to Suicide vs. Homicide \nby Races (Percentages)", x = "Races", y = "Number of Deaths") +
+        theme(plot.title = element_text(hjust = 0.5))
 ```
+By looking at percentages based on this graph, ~ 85% whites who are killed by guns die because of suicide. Meanwhile, about ~ 85% blacks and ~ 70% Hispanics get killed by guns due to homicide.
 
 
 ### Are police-involved gun deaths significantly different from other gun deaths? Assess the relationship between police involvement and age, police involvement and race, and the intersection of all three variables.
 
-```{r police}
+```{r police and age}
+police_gun_deaths <- gun_deaths %>%
+    select(police, age, race) %>%
+    filter(age != "Not Available") %>%
+    mutate(age = as.numeric(age)) %>%

```

Comment:
I like the use of the mutate function here, I did not use that in my code this week but I'll try to find a use for it next week!